### PR TITLE
added ua checking to populate platform column

### LINKED ISSUE #433: add platform from ua
recent fix for lost bat will be expanded to more platforms than android. need to create a column on bat_loss_ table to be filled with a platform category derived from the user agent.

### PR DESCRIPTION
**Pull request by maikelmclauflin**
fixes #433

### CODE REVIEW

File: promotion/controllers.go [hunk 1]
```diff
@@ -562,12 +565,12 @@ func PostReportWalletEvent(service *Service) handlers.AppHandler {
 			walletID,
 			reportID,
 			req.Amount,
+			platform,
 		)
 		if err != nil {
-			if err == errorutils.ErrConflictBATLossEvent {
+			if strings.Contains(err.Error(), errorutils.ErrConflictBATLossEvent) {
```
Comment (husobee) [inline promotion/controllers.go:571]: if errors.Is(err, errorutils.ErrConflictBATLossEvent)  instead?

File: utils/errors/errors.go [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
 
 import (
-	"errors"
 	"fmt"
 )
 
 var (
 	// ErrConflictBATLossEvent is an error created when trying to update a bat loss event with a different amount
-	ErrConflictBATLossEvent = errors.New("unable to update bat loss events")
+	ErrConflictBATLossEvent = "unable to update bat loss events"
```
Comment (husobee) [inline utils/errors/errors.go:9]: Maybe make this back into an error, so we can use errors.Is and errors.As upstream?

File: utils/useragent/useragent.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package useragent
+
+import (
+	"github.com/varstr/uaparser"
```
Comment (evq) [inline utils/useragent/useragent.go:4]: I noticed this parser isn't actively maintained - is there an advantage over something like https://github.com/mssola/user_agent ?

File: utils/useragent/useragent.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package useragent
```
Comment (evq) [inline utils/useragent/useragent.go:1]: Can we add tests for this wrapper using samples of the user agent for Brave on each platform?

File: utils/useragent/useragent.go (hunk 3 above)
Comment (maikelmclauflin) [reply utils/useragent/useragent.go:4]: oh, no i can swap that too

File: utils/useragent/useragent.go (hunk 4 above)
Comment (maikelmclauflin) [reply utils/useragent/useragent.go:1]: fixed in: 49fdac0

File: utils/useragent/useragent.go (hunk 3 above)
Comment (maikelmclauflin) [reply utils/useragent/useragent.go]: fixed in: 49fdac0

**Pull request merged**